### PR TITLE
OTC-408: Create contribution based on GePG payment confirmation and activate policy

### PR DESCRIPTION
**Pull request by hirensoni913**
https://openimis.atlassian.net/browse/OTC-408

### CODE REVIEW

File: OpenImis.RestApi/wwwroot/SentMessages/PaymentConfirmationSms_255744555686353670.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"Message":"Payment has been made to CHF- Invoice: 995940004307 Amount: 1780 TZS Receipt: 92116200000xx938 Date: 28/09/2021 15:58:31 Through:1623416311901","Recipient":"255744555686","Response":"\n{\"error\":false,\"statusCode\":48200,\"statusMessage\":\"Success\",\"data\":[\"Message accepted delivery,Balance:927065\"]}____OK"}]
```
Comment (dragos-dobre) [inline OpenImis.RestApi/wwwroot/SentMessages/PaymentConfirmationSms_255744555686353670.json:1]: All messages from SendMessages to be removed from PR. Add an exception in gitignore.

File: OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs [hunk 2]
```diff
@@ -568,6 +568,97 @@ public int ControlNumbersToBeRequested(string productCode)
...
+                            AND PL.ValidityTo IS NULL
+
+                            IF @TotalPremium >= @PolicyValue
+                            BEGIN
+	                            UPDATE PL SET PolicyStatus = 2, EffectiveDate = dt.PayDate
+	                            FROM tblPolicy PL 
+	                            INNER JOIN @tblPremiums dt ON PL.PolicyID = dt.PolicyId
+	                            WHERE PL.ValidityTo IS NULL;
+
+	                            INSERT INTO tblInsureePolicy(InsureeId, PolicyId, EnrollmentDate, StartDate, EffectiveDate,ExpiryDate, ValidityFrom, AuditUserId, isOffline)
```
Comment (dragos-dobre) [inline OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs:623]: Isn't this already added when synchronizing the family?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs (hunk 2 above)
Comment (hirensoni913) [reply OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs:623]: it's added only if the policy is active and premium exists. In our case there will be no premium and policy will be idle

File: OpenImis.RestApi/wwwroot/SentMessages/PaymentConfirmationSms_255744555686353670.json (hunk 1 above)
Comment (hirensoni913) [reply OpenImis.RestApi/wwwroot/SentMessages/PaymentConfirmationSms_255744555686353670.json:1]: Done!

**Review by dragos-dobre — APPROVED** [review]

**Pull request merged**